### DISCUSSION TITLE
Different response for same question from ChatGPT and LibreChat

### DISCUSSION THREAD

**Discussion by balajeek** [danny-avila/LibreChat#1271] [Q&A]
I use openai chat for many reasons and one major is help with my code (either work or personal)

I asked this question on LibreChat and did not like the response, whereas on ChatGPT i know how it will help me with response and i got what i want, can you tell me why is it different and how can i make it so i get the same response as openai so librechat can be my main resource.

My question was:
"On node-red function i get this warning:
The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.(2362)
const targetTime: Date"

LibreChat response:
`It looks like you're trying to use a Date type as the left-hand side of an arithmetic operation. To fix this, make sure that the left-hand side of the operation is either a number, a BigInt, an enum type or an any type.`

ChatGPT response:
![image](https://github.com/danny-avila/LibreChat/assets/19213088/17d8e060-21c5-40fa-958b-976868efc77a)

**Comment by danny-avila** [comment]
Part of the reason for this is that using LibreChat without any preset/conversation configuration will always be different from ChatGPT.

You can at the very least add ChatGPT's [default system prompt](https://github.com/LouisShark/chatgpt_system_prompt/blob/main/prompts/gpt-4-gizmo-20231116.md) as a prompt prefix (custom instructions) and see how it goes: 
> You are a ChatGPT, a large language model trained by OpenAI, based on the GPT-4 architecture.
Knowledge cutoff: 2023-04
Current date: 2023-11-16
Image input capabilities: Enabled

This is only the tip of the iceberg as it is loaded with other prompts in different contexts. For example, when it generates images, a special prompt is injected instructing it how to refer to the image (this is separate from the instructions on how to generate the image)

Do not doubt that there is careful optimization, both in temperature, presence penalty, and other LLM settings, as well as careful prompt engineering, for ChatGPT. It has to stay competitive over the offerings via API. and may even operate from a special fine-tuned model(s).

The beauty of using the API, though, is customizing it to your liking, and finding something that is satisfactory, even more satisfactory than ChatGPT, that is carefully tailored to your needs; this is definitely possible and has been my own experience.

With the upcoming Assistants API integration that I'm actively working on, I'm aiming to create a default assistant that works as much like ChatGPT as possible. And with these settings, you should be able to create your own assistants that function under a very similar framework as ChatGPT, much more than presently

**Reply by balajeek** [reply]
I didn't even start doing anything about plugin yet, so not concerned about generate images at all (which i may not use it at all)
After i posted this question, i have been asking few more questions and all of the responses are just couple of line sentences and more generic, which is different from what i have been using on chatGPT till now.

What kind of optimization i should set to get same result as chatGPT (atleast for now, before going even enhance further)?

**Reply by danny-avila** [reply]
@balajeek Try these settings 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/ba19e1ec-9589-4903-a50e-7cdab174e066)

You can also import this preset I made via the presets menu:
![image](https://github.com/danny-avila/LibreChat/assets/110412045/0adeac4a-ac97-4bd8-a3be-7812e01d7fc4)

[ChatGPT.json](https://github.com/danny-avila/LibreChat/files/13549042/ChatGPT.json)

Note that the responses are very similar, also note how ChatGPT responds much slower than the API at peak times

_ChatGPT on left, LibreChat on right_

https://github.com/danny-avila/LibreChat/assets/110412045/d5ae9627-0c09-4af4-a4f5-e0b010e9f914

**Reply by danny-avila** [reply]
another example with the preset first:
> Design a database schema for an online merch store.
![image](https://github.com/danny-avila/LibreChat/assets/110412045/0a6b0c44-8d11-4230-927f-957bf752b751)

> Same prompt, without a preset. Note it is much less "chatty"
![image](https://github.com/danny-avila/LibreChat/assets/110412045/393165cd-0848-4e90-bc13-81279a02bcbb)

**Comment by balajeek** [comment]
I tried to import and clicked on it after import, it said 'Preset Active' 
But i don't see 'gpt-4' (in the dropdown next to OpenAI logo) like your screenshot.  Is it that I don't have gpt-4 ?

All along its been 'text-davinci-003' which was the default after i setup LibreChat, I think.

**Reply by danny-avila** [reply]
I'm unaware of your setup, it's possible you don't depending on how you're accessing OpenAI models. If you are still in the free trial tier, you might not

**Reply by balajeek** [reply]
Yes, I am with the free account, not the plus. and i setup my CC for API use on usage basis.

**Reply by danny-avila** [reply]
you might find this helpful/amusing:
![image](https://github.com/danny-avila/LibreChat/assets/110412045/eb4a8caa-0ee9-44c3-890d-d904925a3477)
[source](https://x.com/voooooogel/status/1730726744314069190?s=20)

**Comment by balajeek** [comment]
I changed from  'text-davinci-003'  to 'gpt-3.5-turbo' and now i get the result which is more like the chatGPT
I can live with that.  Thank you.
